addons/resource/models/resource.py

```diff
@@ -221,14 +220,23 @@ def _attendance_intervals(self, start_dt, end_dt, resource=None):
                 until = min(until, attendance.date_to)
             weekday = int(attendance.dayofweek)
 
-            for day in rrule(DAILY, start, until=until, byweekday=weekday):
+            for day in self._rrule(DAILY, start, until, weekday):
                 # attendance hours are interpreted in the resource's timezone
-                dt0 = tz.localize(combine(day, float_to_time(attendance.hour_from)))
-                dt1 = tz.localize(combine(day, float_to_time(attendance.hour_to)))
+                dt0 = self._tz_localize(tz, day, attendance.hour_from)
+                dt1 = self._tz_localize(tz, day, attendance.hour_to)
                 result.append((max(start_dt, dt0), min(end_dt, dt1), attendance))
 
         return Intervals(result)
 
+    @lru_cache(maxsize=3200)
+    def _tz_localize(self, tz, day, hour):
+        combine = datetime.combine
+        return tz.localize(combine(day, float_to_time(hour)))
+
+    @lru_cache(maxsize=4200)
+    def _rrule(self, freq, start, until, weekday):
+        return rrule(freq, start, until=until, byweekday=weekday, cache=True)
```

Comment:
Those things have nothing to do with the model: `self` is not used at all.
I suggest to move them out of the class, so they can be simple functions.
I also suggest to use more suggestive names like `memo_localize` and `memo_rrule`.

I also spot some danger here: `cache=True` on `rrule`.  It means that the cached `rrule` object will eat up memory, but how much?  Can this cache lead to the process using more memory than expected?

What about the cache sizes?  They seem to be tuned to *one* use-case.
What happens if you reduce the cache size?  Is the caching still effective?

Reply:
After a further analysis on the `maxsize` values I did decrease them.

The choice of this values are explained in the following Excel file:
https://drive.google.com/file/d/1vchulanfkMBn0_z8nXgAFUtEhSYychCg/view?usp=sharing
( 2 pages, one for `memo_tz_localize` and the second for `memo_rrule`)
Most of the values are dependent on the databases datas. So I did put some "reasonable realist worst case scenario" (more details on the text part below the Excel Tables).

With this values, in my particular testing case the hit ratio for both methods at the end of the operation are:
 - `memo_tz_localize`:  hit ratio : 95%
 - `memo_rrule`: hit ratio: 71%
( the total execution time increate of ~4s because of this changes)


As the new values are quite small I think we can overlook the `rrule(..., cache=True)`. Furthermore, I don't know how to measure the extra.